### PR TITLE
move from rjsonio and rpython to reticulate

### PR DESCRIPTION
**Pull request by kerschke**
rpython is deprecated and thus the interface for mpm2 needed to be updated

### CODE REVIEW

File: R/sof.mpm2.R [hunk 1]
```diff
@@ -60,25 +61,31 @@ makeMPM2Function = function(n.peaks, dimensions, topology, seed, rotated = TRUE,
   # build parameter set (bounds are [0, 1]^d)
   par.set = makeNumericParamSet("x", len = dimensions, lower = 0, upper = 1)
 
-  # import rPython namespace
-  BBmisc::requirePackages("_rPython", why = "smoof::makeMultiplePeaksModel2Function")
+  # import reticulate namespace
+  BBmisc::requirePackages("_reticulate", why = "smoof::makeMultiplePeaksModel2Function")
+
+  # FIXME: initialize 3 functions from mpm2.py as NULL such that they have a visible binding when checking the pkg
```
Comment (jakobbossek) [inline R/sof.mpm2.R:67]: What is this FIXME about?
As far as I understand the following lines actually achieve what is described here. Hence, it is no FIXME anymore? :-)

File: R/sof.mpm2.R [hunk 2]
```diff
@@ -60,25 +61,31 @@ makeMPM2Function = function(n.peaks, dimensions, topology, seed, rotated = TRUE,
...
+  evaluateProblem = getGlobalOptimaParams = getLocalOptimaParams = NULL
 
   # load funnel generator to global environemt
-  eval(rPython::python.load(system.file("mpm2.py", package = "smoof")), envir = .GlobalEnv)
+  eval(reticulate::py_run_file(system.file("mpm2.py", package = "smoof")), envir = .GlobalEnv)
+  topol = topology
```
Comment (jakobbossek) [inline R/sof.mpm2.R:72]: `topol` is never used after this assignment.

File: R/sof.mpm2.R [hunk 3]
```diff
@@ -60,25 +61,31 @@ makeMPM2Function = function(n.peaks, dimensions, topology, seed, rotated = TRUE,
...
+  eval(reticulate::source_python(system.file("mpm2.py", package = "smoof"), envir = .GlobalEnv, convert = TRUE), envir = .GlobalEnv)
+
+  # extract local and global optima
+  local.opt.params = eval(getLocalOptimaParams(n.peaks, dimensions, topology, seed, rotated, peak.shape))
+  local.opt.params = do.call(rbind, lapply(local.opt.params, unlist))
 
-  local.opt.params = eval(rPython::python.call("getLocalOptimaParams", n.peaks, dimensions, topology, seed, rotated, peak.shape))
-  if (n.peaks == 1)
```
Comment (jakobbossek) [inline R/sof.mpm2.R:70]: Why is this dropped? I guess we should keep this list-conversion if `n.peaks=1`, right?

**Review by jakobbossek — CHANGES_REQUESTED** [review]
Thanks Pascal! I have been successfully procrastinating these simple changes for months 😅
Please address the few raised issues ASAP. Once these issues are solved I am happy to merge.

File: R/sof.mpm2.R (hunk 1 above)
Comment (kerschke) [reply R/sof.mpm2.R]: good point, probably this was still there to check whether this addition was actually needed..

File: R/sof.mpm2.R (hunk 2 above)
Comment (kerschke) [reply R/sof.mpm2.R]: good point, I removed it in the updated version

File: R/sof.mpm2.R (hunk 3 above)
Comment (kerschke) [reply R/sof.mpm2.R:70]: it's not needed anymore.. maybe rPython returned the results in a different format.

**Comment by jakobbossek** [comment]
Out of curiosity I did a very minor benchmark of MPM vs Ackley. Unfortunately I cannot test the old rPython implementation anymore. I believe you have more experience in running experiments with the multiple peaks model smoof interface? Are the results below in line with your experience?

```r
x = matrix(runif(1e5), ncol = 2L)

mpm = makeMPM2Function(dimensions = 2, n.peaks = 1, seed = 123, topology = "funnel")
ack = makeAckleyFunction(dimensions = 2)

print(system.time({apply(x, 1, mpm)}))
print(system.time({apply(x, 1, ack)}))
```

```bash
       User      System verstrichen
      6.412       0.058       6.493
       User      System verstrichen
      0.720       0.003       0.726
```

**Comment by jakobbossek** [comment]
> Out of curiosity I did a very minor benchmark of MPM vs Ackley. Unfortunately I cannot test the old rPython implementation anymore. I believe you have more experience in running experiments with the multiple peaks model smoof interface? Are the results below in line with your experience?
> 
> ```r
> x = matrix(runif(1e5), ncol = 2L)
> 
> mpm = makeMPM2Function(dimensions = 2, n.peaks = 1, seed = 123, topology = "funnel")
> ack = makeAckleyFunction(dimensions = 2)
> 
> print(system.time({apply(x, 1, mpm)}))
> print(system.time({apply(x, 1, ack)}))
> ```
> 
>  ```shell
>        User      System verstrichen
>       6.412       0.058       6.493
>        User      System verstrichen
>       0.720       0.003       0.726
> ```

OK. According to our Mattermost chat the slow execution is not an issue of the new implementation.

**Pull request merged**